browser/ui/webui/settings/pin_shortcut_handler.cc

```diff
@@ -38,20 +38,63 @@ void PinShortcutHandler::HandleCheckShortcutPinState(
     const base::Value::List& args) {
   AllowJavascript();
 
+  CheckShortcutPinState(false);
+}
+
+void PinShortcutHandler::CheckShortcutPinState(bool from_timer) {
   shell_integration::IsShortcutPinned(
       base::BindOnce(&PinShortcutHandler::OnCheckShortcutPinState,
-                     weak_factory_.GetWeakPtr()));
+                     weak_factory_.GetWeakPtr(), from_timer));
 }
 
+#if BUILDFLAG(IS_WIN)
+void PinShortcutHandler::OnPinStateCheckTimerFired() {
+  CheckShortcutPinState(true);
+}
+#endif
+
 void PinShortcutHandler::OnPinShortcut(bool pinned) {
   DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
 
+#if BUILDFLAG(IS_WIN)
+  // On Windows, ignore |pinned| value as Windows asks to user via another
```

Comment:
I'm curious if we still need this when |!pinned|. Does Windows prompt a dialog when unpinning shortcut from taskbar?